src/components/mes-signature-pad/mes-signature-pad.ts

```diff
@@ -0,0 +1,63 @@
+import { Component, ViewChild, EventEmitter, Output } from '@angular/core';
+import { SignaturePad } from 'angular2-signaturepad/signature-pad';
+
+@Component({
+  selector: 'mes-signature-pad',
+  templateUrl: 'mes-signature-pad.html',
+})
+export class MesSignaturePadComponent {
+  @Output() changedDataEvent = new EventEmitter<string>();
```

Comment:
Any reason why we can't dispatch to the store directly from here? Output event emitters can get horribly unwieldy..

Reply:
I was trying to keep the implementation pure, for maximum potential reuse for this component. events can be quickly pushed from the parent to the store.

Reply:
Ah ok, I get the idea. In my mind the store is a fundamental aspect underpinning the application so it wouldn't be impure. How abouts the parent passes down the actions it wants this component to dispatch to?
Don't want to come across pedantic! We made a decision early on not to touch angular's event emitters when we chose to run all our state management through ngrx. The idea was for the store to be the single source of truth for all in-flux app state.